### DISCUSSION TITLE
Inserting post-list directive into template based shortcode

### DISCUSSION THREAD

**Discussion by JakubRakus** [getnikola/nikola#3540] [Q&A]
I have the following situation:

I've created a template based shortcode, say `my_short.tmpl` in `/shortcodes`. This works fine, I can use it in rst files using `{{% my_short %}}`.

I've also created customized `post_list_directive.tmpl` in `/themes/custom/assets/templates`. This also works fine when I use 
```
.. post-list::
   :stop: 3
```
in my rst files.

I'd like to insert my customized post-list into my shortcode template, preferably with ability to pass stop parameter to post-list from parameters of the shortcode, say `{{% my_short stop=3 %}}`.

Is this possible? I've searched through docs but didn't find any suggestions how to invoke a directive like post-list from shortcode tmpl file.

**Comment by Kwpolska** [comment]
`post-list` also works as a shortcode, but I’m not sure if we support nesting shortcodes.

**Reply by JakubRakus** [reply]
It's not good news :( So I assume I have to make some ugly workaround, and paste my shortcode into rst file as a two part raw html and post-list directive between these two parts...

If there are any 'better' solutions I'll submit feature request for nested shortcodes.

**Comment by ralsina** [comment]
There is no reason why nested shortcodes would not work except "never
thought of it"

**Reply by JakubRakus** [reply]
So, what is the proper syntax of shortcode inside tmpl file?
